### PR TITLE
Minor docs fixes

### PR DESCRIPTION
**Pull request by bettinaheim**
Adding link to NGC, refer to GitHub for building from source instructions, and fix https://github.com/NVIDIA/cuda-quantum/issues/27.

### CODE REVIEW

File: docs/sphinx/install.rst [hunk 1]
```diff
@@ -106,69 +110,10 @@ or run the Python examples using the Python interpreter.
...
-                -DLLVM_DIR=/path/to/llvm/lib/cmake/llvm 
-                -DCUDAQ_ENABLE_PYTHON=TRUE
-                \# (optional, if cuquantum is installed)
-                -DCUSTATEVEC=/opt/nvidia/cuquantum
-    ninja install
-    ctest 
+For more information about building CUDA Quantum from source, 
+we refer to the `CUDA Quantum GitHub repository`_.
+
+.. _CUDA Quantum GitHub repository: https://github.com/NVIDIA/cuda-quantum
```
Comment (boschmitt) [inline docs/sphinx/install.rst:116]: Needs to add these instructions to the repository if we are telling users to go looking for them here.

Comment (bettinaheim) [reply docs/sphinx/install.rst:116]: They are in the repository.

Comment (boschmitt) [reply docs/sphinx/install.rst:116]: My bad, I missed the `Building.md` in the repo.

Comment (boschmitt) [inline docs/sphinx/install.rst:116]:
```suggestion
.. _CUDA Quantum GitHub repository: https://github.com/NVIDIA/cuda-quantum/blob/main/Building.md
```

**Review by boschmitt — APPROVED** [review]

**Review by schweitzpgi — APPROVED** [review]

**Pull request merged**